criu/pie/restorer.c

```diff
@@ -202,11 +202,33 @@ static int restore_creds(struct thread_creds_args *args, int procfd,
 	 * Setup supplementary group IDs early.
 	 */
 	if (args->groups) {
-		ret = sys_setgroups(ce->n_groups, args->groups);
-		if (ret) {
-			pr_err("Can't setup supplementary group IDs: %d\n", ret);
+		// We may be in an unprivileged user namespace where we're not
+		// allowed to call setgroups. If the current list of groups is already
+		// what we want, skip the call to setgroups.
+
+		int n = sys_getgroups(0, NULL);
```

Comment:
`sys_getgroups()` may fail.
Please add a check for its return value. This might require some adjustments to the declaration block as we generally don't mix declarations and code.

Reply:
According to man, it cannot fail if `size` is 0. Unless we want to take into account future changes to getgroups.

Reply:
I don't think it'd hurt to provide an additional check in the case that getgroups does get modified.